R/theme.R

```diff
@@ -34,14 +34,12 @@ data_deps_path <- function(pkg) {
 bs_theme <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
 
-  theme <- get_bootswatch_theme(pkg)
-  theme <- check_bootswatch_theme(theme, pkg$bs_version, pkg)
+  bs_theme_args <- pkg$meta$template$bslib %||% list()
+  bs_theme_args[["version"]] <- pkg$bs_version
+  bs_theme_args[["preset"]] <- get_bslib_theme(pkg)
```

Comment:
We're very close to allowing `template.bslib` carry the entire theme spec. For example,

```yaml
template:
  bootswatch: darkly
  bootstrap: 5
  bslib:
    body-color: darkslategray
```

could theoretically be

```yaml
template:
  bslib:
    version: 5
    preset: darkly
    body-color: darkslategray
```

except for some friction around `version` (`template.bslib.version` is overwritten by `template.bootstrap`) and the `bootswatch` keyword (including `template.bslib.bootswatch` will cause bslib to throw an error [only one of `preset` or `bootswatch` is allowed and we use `preset` in this PR]).

If desirable, I could smooth over those two edge cases.

Reply:
I think that would be great!

Reply:
Will do! @hadley do you recall off-hand the difference between `pkg$meta$template$bootswatch` and `pkg$meta$template$params$bootswatch`. In particular, I'm wondering if I should consult these together, i.e. if not one then the other, or are the `$params$bootswatch` inherently lower priority, e.g. if `$params` comes from a global template.

Reply:
I don't 😞

Reply:
I think it's historical: some things that went under `template.params` are now available directly under `template` (https://github.com/r-lib/pkgdown/pull/1590). Although there's still some ambiguity: https://github.com/r-lib/pkgdown/issues/2182.

`bootswatch` is one of the parameters that was unnested, explaining why pkgdown looks at `template.bootswatch` and `template.params.bootswatch`. I can't find any evidence that `bslib` was ever expected to be found under `params`.

Reply:
Sure... I held off from doing that because I wasn't sure how strongly you'd want to recommend putting theming-related options under `bslib`. What do you have in mind?

Reply:
It feels more consistent to me to just put everything under bslib (with the possible exception of version, since it's nice to have a way at the top-level to opt-in to bs5)

Reply:
@hadley I proposed updates to the customize vignette in 445d5ae94cc1ea132b21d3fd8cc870c1e2f10ea4. Please let me know if you need anything else!